src/WebJobs.Script.WebHost/Filters/AuthorizationLevelAttribute.cs

```diff
@@ -100,6 +101,26 @@ internal static AuthorizationLevel GetAuthorizationLevel(HttpRequestMessage requ
             return AuthorizationLevel.Anonymous;
         }
 
+        internal static bool IsValidKey(string keyToValidate, string functionKey)
+        {
+            if (string.IsNullOrEmpty(functionKey) ||
+                string.IsNullOrEmpty(keyToValidate))
+            {
+                return false;
+            }
+
+            string[] keys = functionKey.SplitAndTrim(',');
+            foreach (string key in keys)
+            {
```

Comment:
This screams for some Linq love :)

    return keys.Any(k => SecretEqual(keyToValidate, k));